### PR TITLE
Minor explosives rework

### PR DESCRIPTION
**Pull request by SamaelGray**
## Changes

- Made secondary bomb and burn damage (essentially ammo payload) solely dependent on projectile mass, This change does not impact charged and plasma ammo.
- Changed the formula used for HEDP bullet damage calculations in the projectile spreadsheet to be similar to HEAT ammo and made the necessary changes in xml.  No more hacky ways of getting the numbers to work.
- Minor tweaks, fixes and housekeeping.

## References

- https://docs.google.com/spreadsheets/d/1P9U8EtYoRBh-jPMPmXcqam1O3qvKZPMml2ktAMPssOk/edit#gid=1393347070

## Reasoning

- Secondary damages were directly derived from muzzle energy which makes no sense, it made heavy and higher caliber rounds with low velocity get unreasonably low payload damage values, also made low velocity and short barrel version of the same ammo deal less secondary damage. Most of the common ammo types received minimal to no change as the result of how the function is altered.
- HEAT and HEDP are pretty much the same thing.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without error

### CODE REVIEW

File: Defs/Ammo/Generic/LauncherGrenade.xml [hunk 1]
```diff
@@ -50,12 +51,26 @@
...
 		</statBases>
 		<ammoClass>GrenadeHE</ammoClass>
 		<detonateProjectile>Bullet_40x46mmGrenade_HE</detonateProjectile>
 	</ThingDef>
 
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="AmmoLauncherGrenadeBase">
+		<defName>Ammo_LauncherGrenade_HE_TFuzed</defName>
+		<label>launcher grenade (HE Time-Fuzed)</label>
+		<graphicData>
+			<texPath>Things/Ammo/STACK_RANGED/GrenadeLauncher/AIR</texPath>
```
Comment (N7Huntsman) [inline Defs/Ammo/Generic/LauncherGrenade.xml:64]: This texture doesn't exist.

File: Defs/Ammo/HighCaliber/338LapuaMagnum.xml [hunk 2]
```diff
@@ -132,7 +132,7 @@
 		<label>.338 Lapua Magnum bullet (FMJ)</label>
 		<projectile Class="CombatExtended.ProjectilePropertiesCE">
 			<damageAmountBase>26</damageAmountBase>
-			<armorPenetrationSharp>11</armorPenetrationSharp>
```
Comment (N7Huntsman) [inline Defs/Ammo/HighCaliber/338LapuaMagnum.xml:135]: Unless you've got a compelling source to cite, leave it--from the good sources I've got and some napkin math, we're already being _very_ generous w/ .338 LM FMJ's sharp pen.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]

**Comment by N7Huntsman** [comment]
I had a couple of paragraphs of other feedback written out, but Github apparently ate it, so I'll try to remember and write it out below...

**Comment by N7Huntsman** [comment]
1. The secondary damage factor of AP-I and AP-HE have been increased from 50% and 60% to 100% and 150%, respectively. Why?

2. I'm not sure how exactly you implemented the 25% bomb damage reduction to HEDP and HEAT (hard to find it among the other formula changes), but it's probably most straightforward to just have secondary damage include the secondary damage factor from the Table value, by changing the Secondary Damage Calcs formula from `=IF(VLOOKUP(B6, AmmoTypeFactors, 6, FALSE) = "Bomb", J6, C6 * VLOOKUP(B6, AmmoTypeFactors, 4, FALSE))` to `=IF(VLOOKUP(B588, AmmoTypeFactors, 6, FALSE) = "Bomb", J588, C588) * VLOOKUP(B588, AmmoTypeFactors, 4, FALSE)`.

3. You removed velocity as a factor in secondary damage for blast and incendiary damage types. Should we consider implementing this for charged ammo blast/emp effects as well?

4. Regarding the new blast damage formula `y = 5.15 * ( x^0.925 )` , it doesn't intersect with the old formula `y = x * 10 / 3` until around x = 330. This means any bombs with Pressure (Explosion) of < 330 get a damage increase and any will > 330 get a damage decrease. However, there's only 1 projectile on the sheet (the Tomahawk, an outlier) that has Pressure (Explosion) > 330. This means _every_ other bombs gets its damage buffed; +5% for the big stuff like 155mm, and +20% for the more typical stuff like 40x46mm grenades.
I recommend we reduce the 5.15 to something closer to 4, so that the two formulas intersect a lot sooner, in the vicinity of the average Pressure (Explosive)--around 25 with the Tomahawk removed--or slightly above, so that it has an equalizing effect--smaller bombs get their damage buffed, bigger ones get their damage reduced.

File: Defs/Ammo/HighCaliber/338LapuaMagnum.xml (hunk 2 above)
Comment (SamaelGray) [reply Defs/Ammo/HighCaliber/338LapuaMagnum.xml:135]: Yea there is real world testing done and .338 Lapua penetrates half an inch of mild steel plate (commonly used for CE's AP values), linked in the spreadsheet. Can lower to 11.5 instead or nerf .338 Norma by 0.5 as it's a heavier and slower round delivering less energy overall.

**Comment by N7Huntsman** [comment]
Merging this into a side branch to cherry-pick some of the changes for the initial 1.5 release. The various housekeeping changes are fine, but I'm still working over the projectile sheet, so some the actual stat change stuff is liable to change again before it's all said and done--easier to preserve your branch to point at dev again later rather than recreating it/backing changes out.

**Pull request merged**